build_tools/cmake/setup_build.sh

```diff
@@ -16,6 +16,8 @@ CMAKE_BIN="${CMAKE_BIN:-$(which cmake)}"
 ninja --version
 python3 --version
 
+IREE_PYTHON3_EXECUTABLE="${IREE_PYTHON3_EXECUTABLE:-$(which python3)}"
```

Comment:
Not needed in this PR, just brainstorming...

We have enough of these scripts and env vars following the same structure now that it might help to document their usage in build_tools/cmake/README.md in addition to (or instead of) the comments in each file.

Stuff to document:
* That our CI runs these scripts
* That the scripts can be run manually by developers
* What prerequisites the scripts have (and how Docker images can help with that, if you want)
* How the scripts can be configured via environment variables (overriding the inferred defaults)